interface/batchcom/batchcom.php

```diff
@@ -24,280 +29,217 @@
-        }
-        if ($_POST['app_e']!=0 AND $_POST['app_e']!='') {
-            $and=where_or_and ($and);
-            $sql_where_a.=" $and cal_events.pc_endDate < '".$_POST['app_e']."'";
-        }
-        $sql.=$sql_where_a;
-
+        if ($_POST['app_s']!=0 and $_POST['app_s']!='') { $sql .= " and cal_events.pc_eventDate >= '".$_POST['app_s']."'"; }
+        if ($_POST['app_e']!=0 and $_POST['app_e']!='') { $sql .= " and cal_events.pc_endDate <= '".$_POST['app_e']."'"; }
         // encounter dates
-        if ($_POST['seen_since']!=0 AND $_POST['seen_since']!='') {
-            $and=where_or_and ($and);
-            $sql.=" $and forms.date > '".$_POST['seen_since']."' " ;
-        }
-        if ($_POST['seen_upto']!=0 AND $_POST['not_seen_since']!='') {
-            $and=where_or_and ($and);
-            $sql.=" $and forms.date > '".$_POST['seen_since']."' " ;
-        }
-
+        if ($_POST['seen_since']!=0 and $_POST['seen_since']!='') { $sql .= " and forms.date >= '".$_POST['seen_since']."' " ; }
+        if ($_POST['seen_before']!=0 and $_POST['seen_before']!='') { $sql .= " and forms.date <= '".$_POST['seen_before']."' " ; }
         // age
-        if ($_POST['age_from']!=0 AND $_POST['age_from']!='') {
-            $and=where_or_and ($and);
-            $sql.=" $and DATEDIFF( CURDATE( ), patient_data.DOB )/ 365.25 >= '".$_POST['age_from']."' ";
-        }
-        if ($_POST['age_upto']!=0 AND $_POST['age_upto']!='') {
-            $and=where_or_and ($and);
-            $sql.=" $and DATEDIFF( CURDATE( ), patient_data.DOB )/ 365.25 <= '".$_POST['age_upto']."' ";
-        }
-
+        if ($_POST['age_from']!=0 and $_POST['age_from']!='') { $sql .= " and DATEDIFF( CURDATE( ), patient_data.DOB )/ 365.25 >= '".$_POST['age_from']."' "; }
+        if ($_POST['age_upto']!=0 and $_POST['age_upto']!='') { $sql .= " and DATEDIFF( CURDATE( ), patient_data.DOB )/ 365.25 <= '".$_POST['age_upto']."' "; }
         // gender
-        if ($_POST['gender']!='Any') {
-            $and=where_or_and ($and);
-            $sql.=" $and patient_data.sex='".$_POST['gender']."' ";
-        }
-
+        if ($_POST['gender']!='Any') { $sql .= " and patient_data.sex='".$_POST['gender']."' "; }//INJECTION VECTOR HERE
         // hipaa overwrite
-        if ($_POST['hipaa_choice']!='NO') {
-            $and=where_or_and ($and);
-            $sql.=" $and patient_data.hipaa_mail='YES' ";
-        }
+        if ($_POST['hipaa_choice'] != $hipaa_choices[0]) { $sql .= " and patient_data.hipaa_mail='YES' "; }
 
         switch ($_POST['process_type']):
             case $choices[1]: // Email
-                $and=where_or_and ($and);
-                $sql.=" $and patient_data.email IS NOT NULL ";
+                $sql.=" and patient_data.email IS NOT NULL ";
             break;
         endswitch;
 
         // sort by
-        $sql.=' ORDER BY '.$_POST['sort_by'];
-        //echo $sql;
+        $sql.=' ORDER BY '.$_POST['sort_by'];//INJECTION VECTOR
```

Comment:
btw, here's the escape_identifier function:
https://github.com/openemr/openemr/blob/master/library/formdata.inc.php#L186